src/test/java/org/prebid/server/bidder/invibes/InvibesBidderTest.java

```diff
@@ -0,0 +1,242 @@
+        assertThat(result.getValue()).hasSize(1);
+        assertThat(result.getValue().get(0).getUri()).isEqualTo("https://bid3.videostep.com/test");
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnErrorWhenImpExtCouldNotBeParsed() {
+        // given
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(singletonList(Imp.builder()
+                        .ext(mapper.valueToTree(ExtPrebid.of(null, mapper.createArrayNode())))
+                        .build()))
+                .build();
+
+        // when
+        final Result<List<HttpRequest<InvibesBidRequest>>> result = invibesBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1);
+        assertThat(result.getErrors().get(0).getMessage()).startsWith("Error parsing invibesExt parameters");
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnErrorWhenBannerIsNull() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(identity(), ExtImpInvibes
+                .of("12", 15, InvibesDebug.of("test", true)));
+
+        final BidRequest bidRequestWithoutBanner = bidRequest.toBuilder().imp(
+                Collections.singletonList(bidRequest.getImp().get(0).toBuilder().banner(null).build())).build();
+
+        // when
+        final Result<List<HttpRequest<InvibesBidRequest>>> result =
+                invibesBidder.makeHttpRequests(bidRequestWithoutBanner);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1)
+                .containsOnly(BidderError.badInput("Banner not specified"));
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnErrorWhenSiteIsNotPresent() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(identity(), ExtImpInvibes
+                .of("12", 15, InvibesDebug.of("test", true)));
+
+        // when
+        final Result<List<HttpRequest<InvibesBidRequest>>> result = invibesBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1)
+                .containsOnly(BidderError.badInput("Site not specified"));
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnErrorWhenSiteIsNotPresent2() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(identity(), ExtImpInvibes
+                .of("12", 15, InvibesDebug.of("test", true))).toBuilder()
+                .site(Site.builder().page("www.awesome-page.com").build()).build();
+
+        // when
+        final Result<List<HttpRequest<InvibesBidRequest>>> result = invibesBidder.makeHttpRequests(bidRequest);
+
+        // then
+        final InvibesBidRequest expectedRequest = InvibesBidRequest.builder()
+                .bidParamsJson("{\"PlacementIds\":[\"12\"],\"BidVersion\":\"4\","
```

Comment:
create object and use mapper to turn it into json, 
(in the future it will be a headache to modify raw json)